intro_calculating_cred_grain.md

```diff
@@ -0,0 +1,162 @@
+ 
+![](https://docs.google.com/drawings/u/1/d/s50mxZJ1DFD_yr4OFUxUd9g/image?w=551&h=250&rev=57&ac=1&parent=1kRxnuKAogf_cE0y6u2QYcjQimE0-jrBZ-LxcXKOUITw)
+
+> Figure 1. A very high level overview of how SourceCred allocates Cred and Grain, distinguishing between actions taken by contributors and those taken by the SourceCred algorithm.
+
+The rest of this doc outlines how collaboration data is gathered and structured, and how the exact amounts of Cred and Grain are calculated, and ends with some future directions SourceCred as a project is considering.
+
+## Components of contributing in SourceCred
+
+SourceCred first generates a weights graph from the input data. In this graph, a node is a contributor or contribution, and edges represent some sort of connection between the nodes. These connections are intended to include all possible ways that a contribution or contributor on a platform can express dependencies, thanks, or flows of value. Examples would include anything from a mention on a forum post to liking or merging a PR on GitHub.
+
+#### Nodes:
+
+Contributions are represented as nodes in the graph, and are given addresses to identify them uniquely. These addresses include a prefix, which takes the form: organization making the plugin that generates the node + name of the plugin + type of contribution (e.g. topic/post/user/like).
+
+Contributors are also represented as nodes in the graph. These nodes are similarly prefixed, and connected to contributions with directed edges, as illustrated in the figure below. For example, if an author node creates an issue, the author node gets an incoming edge from the issue node. Or if an author of a Discourse post mentions another contributor in the post, the post creates an outbound edge coming from the author.
+
+Lastly, initiatives are another type of node that serve as a high level place to connect relevant nodes and discuss governance. It's similar to a tracking issue, but generalized across the weights graph, providing a way to organize planned work for a well-scoped task.
+
+#### Weights:
+
+Both nodes and edges have weights, and an edge object in the SourceCred implementation can have forward and backwards weight. We’ll get into what these weights mean shortly, but for now, it’s worth noting that these weights can easily be set by address prefix. Setting by prefix is implemented as a multiplicative update; for example, setting issue comments to be worth twice an issue emoji reaction creates a 2:1 ratio that is preserved by decreasing GitHub’s edge weights by 2 relative to those of Discourse. Additionally, nodes can have self edges, which is relevant when a piece of content is edited, or an contribution references itself.
+  
+![](https://docs.google.com/drawings/u/1/d/sIDpfiaPlACrxNScrfsISkQ/image?w=186&h=230&rev=47&ac=1&parent=1kRxnuKAogf_cE0y6u2QYcjQimE0-jrBZ-LxcXKOUITw)
```

Comment:
I think this can be too abstract for the reader. If we have an ongoing "simple example subgraph" that we establish above, then here we can attach weights to that graph, and the reader can follow along more easily. ALso, worth remembering that almost all graph nodes have a weight of 0, since nonzero weight corresponds to cred minting.

Reply:
Yeah, I think having an ongoing example would be tough.  
I tried to address the low scores in the text, but the example has a Discourse like and a GitHub PR, which are fairly high-scoring.